pkg/alerts/alertsHandler/alertsHandler.go

```diff
@@ -193,34 +193,59 @@ func ProcessSilenceAlertRequest(ctx *fasthttp.RequestCtx) {
 	}
 
 	// Parse request
-	var silenceRequest struct {
-		AlertID        string `json:"alert_id"`
-		SilenceMinutes uint64 `json:"silence_minutes"`
-	}
-	if err := json.Unmarshal(ctx.PostBody(), &silenceRequest); err != nil {
+	var request alertutils.AlertSilenceRequest
+	if err := json.Unmarshal(ctx.PostBody(), &request); err != nil {
 		utils.SendError(ctx, fmt.Sprintf("Failed to unmarshal json. Error=%v", err), "", err)
 		return
 	}
 
-	// Find alert and update SilenceMinutes
-	alertDataObj, err := databaseObj.GetAlert(silenceRequest.AlertID)
+	// Find alert in the database
+	alertDataObj, err := databaseObj.GetAlert(request.AlertID)
 	if err != nil {
-		utils.SendError(ctx, fmt.Sprintf("Failed to find alert. Error=%v", err), fmt.Sprintf("alert ID: %v", silenceRequest.AlertID), err)
+		utils.SendError(ctx, fmt.Sprintf("Failed to find alert. Error=%v", err), fmt.Sprintf("alert ID: %v", request.AlertID), err)
 		return
 	}
 
-	alertDataObj.SilenceMinutes = silenceRequest.SilenceMinutes
-	// Update the SilenceMinutes
+	// Handle Silence or Unsilence based on mode
+	if mode == alertutils.SilenceMode {
+		if request.SilenceMinutes == 0 {
+			utils.SendError(ctx, "SilenceMinutes must be greater than zero", "", nil)
+			return
+		}
+		if request.SilenceEndTime == 0 {
+			utils.SendError(ctx, "SilenceEndTime must be greater than zero", "", nil)
+			return
+		}
+		alertDataObj.SilenceMinutes = request.SilenceMinutes
+		alertDataObj.SilenceEndTime = request.SilenceEndTime
```

Comment:
A few things:
1. Why do both `SilenceMinutes` and `SilenceEndtime` need to be set? Aren't these redundant pieces of information?
2. What happens if `SilenceEndtime != now() + SilenceMinutes`?

Reply:
1. Previously, we only stored the silence duration with `SilenceMinutes`, but it wasn’t easy to calculate how much time was left for the alert to become active again. So, I added `SilenceEndTime` to directly calculate the exact time when the silence ends on the frontend and display it in the UI.

2. I moved the calculation of `SilenceEndTime` to the backend to avoid inconsistencies between `SilenceEndTime` and `now() + SilenceMinutes`.